tsl/src/bgw_policy/job_api.c

```diff
@@ -201,10 +201,24 @@ static BgwJob *
 find_job(int32 job_id, bool null_job_id, bool missing_ok)
 {
 	BgwJob *job;
+	bool got_lock;
+	LOCKTAG tag;
 
 	if (null_job_id && !missing_ok)
 		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE), errmsg("job ID cannot be NULL")));
 
+	/* The job table manipulation functions in job.c use an advisory lock
+	 * rather than a row lock to prevent the job from being modified while
+	 * running. Since we use this function to find a job for either running
+	 * manually or for altering it, we need to grab that advisory lock here as
+	 * well. */
+	if (!(got_lock = ts_lock_job_id(job_id,
+									RowShareLock,
+									/* session_lock */ false,
+									&tag,
+									/* block */ true)))
+		return NULL;
```

Comment:
By the way, when is this null, when the job is modified concurrently? Codecov says it's not covered, maybe there's an easy way to construct an isolation test for this...

Reply:
We're bad on handling these cases (there are several) so not sure how easy it is to create a test case for this, but will take a look if there is an easy way.

Reply:
This particular return can't fire, so we should probably just replace it with an assert. The return value can be `false` if it is a non-blocking wait and the lock is taken, not otherwise.

Reply:
Done.